https://github.com/johang/btfs/pull/29

From 8abf27a7bf6c0d172410f14555f2e9f0862f45f6 Mon Sep 17 00:00:00 2001
From: Juan RP <xtraeme@voidlinux.eu>
Date: Fri, 6 May 2016 07:08:28 +0200
Subject: [PATCH] Check for pthread_setname_np() correctly.

Do not blindly assume that it's available in all OSes but Apple.
Not all C libraries implement this because it's a non POSIX extension.

Fixes compilation with the musl C library.
---
 configure.ac | 3 +++
 src/btfs.cc  | 2 +-
 2 files changed, 4 insertions(+), 1 deletion(-)

diff --git a/configure.ac b/configure.ac
index 15be5df..030c7ff 100644
--- configure.ac
+++ configure.ac
@@ -23,4 +23,7 @@ AC_TYPE_SIZE_T
 # Checks for library functions.
 AC_CHECK_FUNCS([memset mkdir realpath strdup])
 
+# Check for unportable pthread_setname_np()
+AC_CHECK_LIB(pthread, pthread_setname_np)
+
 AC_OUTPUT(Makefile src/Makefile scripts/Makefile)
diff --git a/src/btfs.cc b/src/btfs.cc
index 2d0a434..bbdf6d3 100644
--- src/btfs.cc
+++ src/btfs.cc
@@ -512,7 +512,7 @@ btfs_init(struct fuse_conn_info *conn) {
 	pthread_create(&alert_thread, NULL, alert_queue_loop,
 		new Log(p->save_path + "/../log.txt"));
 
-#ifndef __APPLE__
+#ifdef HAVE_PTHREAD_SETNAME_NP
 	pthread_setname_np(alert_thread, "alert");
 #endif
 
-- 
2.8.2

